travis-docker-push.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+TAG="${TRAVIS_BRANCH}-${TRAVIS_BUILD_NUMBER}-${TRAVIS_COMMIT}"
+echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+docker build -t "thedxw/beis-report-official-development-assistance:$TRAVIS_BRANCH" .
```

Comment:
I noticed we are [building a docker image earlier in the Travis file](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/.travis.yml#L5) for the testing step and wondered if we could consolidate as we are now building 2 images, which take a while. It would also be nice to only push the exact image that was used for testing.

I see that the test image is currently being built with the build-arg that sets `RAILS_ENV=test` which installs test gems we don't want in production. I think we can change that by installing the test gems inside the test container rather than the image with:

```
before_install:
  docker build .  -t app/test
script:
  docker exec test-container bundle --with test
  docker exec test-container rake
```

I wonder if we could break the `build` part out of this script?

Reply:
Yes, I was wondering about the duplication but as you say the refactor can come after this is merged to move things forward.